public/docs/_examples/upgrade-module/ts/app/divide-routes/app.module.ts

```diff
@@ -0,0 +1,62 @@
+// #docregion
+declare var angular: angular.IAngularStatic;
+import { NgModule } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { UpgradeModule } from '@angular/upgrade/static';
+
+import { HeroModule } from './hero.module';
+
+// #docregion router-config
+import { HashLocationStrategy, LocationStrategy } from '@angular/common';
+import { RouterModule, UrlHandlingStrategy, UrlTree } from '@angular/router';
+import { AppComponent } from './app.component';
+
+class Ng1Ng2UrlHandlingStrategy implements UrlHandlingStrategy {
+  // use only process the `/hero` url
+  shouldProcessUrl(url: UrlTree) { return url.toString().startsWith('/hero'); }
+  extract(url: UrlTree) { return url; }
+  merge(url: UrlTree, whole: UrlTree) { return url; }
+}
+
+@NgModule({
+  imports: [
+    BrowserModule,
+    UpgradeModule,
+    HeroModule,
+    RouterModule.forRoot([])
+  ],
+  providers: [
+    // use hash location strategy
+    { provide: LocationStrategy, useClass: HashLocationStrategy },
```

Comment:
Is this needed for upgrade? IIRC the router module had a parameter to include this strategy or not.